### PR TITLE
Add blog about oVirt SAML with keycloak

### PR DESCRIPTION
**Pull request by machacekondra**
This PR add new blog post about SAML integration with oVirt using Keycloak.

### CODE REVIEW

**Comment by machacekondra** [comment]
@mwperina

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+In this part we create last peace which we need to get this up and running.
+
+### Installation
+```
+# yum install -y ovirt-engine-extension-aaa-ldap ovirt-engine-extension-aaa-misc
+```
+
+### Create /etc/ovirt-engine/extensions.d/http-authn.properties
+```
+ovirt.engine.extension.name = http-authn
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: `saml-authn` may be a better name

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+ovirt.engine.extension.bindings.method = jbossmodule
+ovirt.engine.extension.binding.jbossmodule.module = org.ovirt.engine-extensions.aaa.misc
+ovirt.engine.extension.binding.jbossmodule.class = org.ovirt.engineextensions.aaa.misc.http.AuthnExtension
+ovirt.engine.extension.provides = org.ovirt.engine.api.extensions.aaa.Authn
+ovirt.engine.aaa.authn.profile.name = example-http
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: `saml` may be a better name

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+ovirt.engine.aaa.authn.authz.plugin = ldap-authz
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: `saml-authz` may be a better name

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+ovirt.engine.aaa.authn.mapping.plugin = example-http-mapping
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: `saml-http-mapping` may be a better name

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+config.artifact.name = HEADER
+config.artifact.arg = X-Remote-User
+```
+
+### Create /etc/ovirt-engine/extensions.d/example-http-mapping.properties
+```
+ovirt.engine.extension.name = example-http-mapping
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: `saml-http-mapping`

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+ovirt.engine.extension.provides = org.ovirt.engine.api.extensions.aaa.Mapping
+config.mapAuthRecord.type = regex
+config.mapAuthRecord.regex.mustMatch = false
+config.mapAuthRecord.regex.pattern = ^(?<user>.*?)((\\\\(?<at>@)(?<suffix>.*?)@.*)|(?<realm>@.*))$
+config.mapAuthRecord.regex.replacement = ${user}${at}${suffix}
+```
+
+### Create /etc/ovirt-engine/extensions.d/ldap-authz.properties
+```
+ovirt.engine.extension.name = ldap-authz
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: `saml-authz`

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 7]
```diff
@@ -0,0 +1,185 @@
...
+ovirt.engine.extension.bindings.method = jbossmodule
+ovirt.engine.extension.binding.jbossmodule.module = org.ovirt.engine-extensions.aaa.ldap
+ovirt.engine.extension.binding.jbossmodule.class = org.ovirt.engineextensions.aaa.ldap.AuthzExtension
+ovirt.engine.extension.provides = org.ovirt.engine.api.extensions.aaa.Authz
+config.profile.file.1 = ../aaa/ldap.properties
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: `saml-ldap.properties` may be a better name

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 8]
```diff
@@ -0,0 +1,185 @@
...
+```
+
+### Create /etc/ovirt-engine/aaa/ldap.properties
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: saml-ldap.properties

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md [hunk 9]
```diff
@@ -0,0 +1,185 @@
...
+pool.default.serverset.type = single
+pool.default.serverset.single.server = ${global:vars.server}
+```
+
+This is it, now restart ovirt-engine:
+```
+systemctl restart ovirt-engine.service
+```
+
+If you now try to login to webadmin, you will be redirected `Keycloak`, where if you didn't login before you can enter credentials from your 389ds ldap.
```
Comment (mwperina) [inline source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: If you try to login to webadmin, you will be redirected to `Keycloak` (if you are not already logged in), where you can enter credentials from your 389ds ldap setup.

**Comment by gregsheremeta** [comment]
build failing because
https://github.com/tenex/rails-assets/issues/433

File: source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md (hunk 9 above)
Comment (machacekondra) [reply source/blog/2018-08-03-sso-keycloak-ovirt-saml-389ds.html.md]: Done

**Pull request merged**